packages/contracts-rfq/contracts/FastBridgeV2.sol

```diff
@@ -139,8 +139,13 @@ contract FastBridgeV2 is Admin, IFastBridgeV2 {
         if (block.timestamp > transaction.deadline) revert DeadlineExceeded();
 
         // mark bridge transaction as relayed
-        if (bridgeRelays[transactionId]) revert TransactionRelayed();
-        bridgeRelays[transactionId] = true;
+        if (bridgeRelayDetails[transactionId].blockNumber != 0) revert TransactionRelayed();
+        
+        bridgeRelayDetails[transactionId] = BridgeRelay({
+            blockNumber: uint48(block.number),
+            blockTimestamp: uint48(block.timestamp),
+            relayer: relayer
+        });
```

Comment:
This is a trivial change, but we still aim to get these covered in the tests. Both of these would be useful for testing:
- https://book.getfoundry.sh/cheatcodes/warp
- https://book.getfoundry.sh/cheatcodes/roll

Reply:
Also, I just realized that we don't have a unit test for `relayer == address(0)` case. Might as well add this one in destination-specific tests - and we could utilise this invariant in `bridgeRelays`

Reply:
doesnt seem like we're actually restricting relayer != address(0)

I'm adding that to the function as well - let me know if this is not what you had in mind